index.js

```diff
@@ -11,12 +21,24 @@ function isDeprecated(propType, message) {
   };
 }
 
-export default function addIsDeprecated(PropTypes) {
-  for (const type in PropTypes) {
-    if (PropTypes.hasOwnProperty(type)) {
-      const propType = PropTypes[type];
-      PropTypes[type].isDeprecated = isDeprecated.bind(PropTypes, propType);
+/**
+ * Returns a copy of `PropTypes` with an `isDeprecated`
+ * method available on all top-level propType options.
+ * @param {React.PropTypes}  PropTypes
+ * @return {React.PropTypes} newPropTypes
+ */
+export function addIsDeprecated(PropTypes) {
+  let newPropTypes = {...PropTypes};
+  for (const type in newPropTypes) {
+    if (newPropTypes.hasOwnProperty(type)) {
+      let propType = newPropTypes[type];
+      propType = propType.bind(newPropTypes);
+      propType.isDeprecated = deprecate.bind(newPropTypes, propType);
+      newPropTypes = {
```

Comment:
Looks like you can already use assignment here because it has been cloned earlier in the method.